File: Children-Competition-Java/mpp-proiect-java-ABNegrea/Problema5/src/main/java/Concurs/Domain/Event.java
package Concurs.Domain;

import java.util.UUID;

public class Event extends Entity<UUID>{
    private String name;
    private int minAge;
    private int maxAge;

    private int enrolled;

    public Event(String name, int minAge, int maxAge, int enrolled) {
        this.setId(UUID.randomUUID());
        this.name = name;
        this.minAge = minAge;
        this.maxAge = maxAge;
        this.enrolled = enrolled;
    }

    public Event(UUID id, String name, int minAge, int maxAge, int enrolled) {
        this.setId(id);
        this.name = name;
        this.minAge = minAge;
        this.maxAge = maxAge;
        this.enrolled = enrolled;
    }

    public String AgeToString(){
        return String.valueOf(this.minAge) + "-" + String.valueOf(this.maxAge);
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getMinAge() {
        return minAge;
    }

    public void setMinAge(int minAge) {
        this.minAge = minAge;
    }

    public int getMaxAge() {
        return maxAge;
    }

    public void setMaxAge(int maxAge) {
        this.maxAge = maxAge;
    }

    public int getEnrolled() {
        return enrolled;
    }

    public void setEnrolled(int enrolled) {
        this.enrolled = enrolled;
    }
}

File: Children-Competition-C#/Domain/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concurs.Domain
{
    internal class Event : Entity<Guid>
    {
        public string Name { get; set; }
        public int MinAge { get; set; }
        public int MaxAge { get; set; }

        public int Enrolled { get; set; }

        public Event(Guid guid, string n, int min, int max, int enrolled)
        {
            this.Id = guid;
            this.Name = n;
            this.MinAge = min;
            this.MaxAge = max;
            this.Enrolled = enrolled;
        }

        public Event(string n, int min, int max, int enrolled)
        {
            this.Id = Guid.NewGuid();
            this.Name = n;
            this.MinAge = min;
            this.MaxAge = max;
            this.Enrolled = enrolled;
        }

        public string AgeToString()
        {
            return this.MinAge.ToString() + "-" + this.MaxAge.ToString();
        }

        public override string ToString()
        {
            return this.Name;
        }

    }
}
